5 m1 checkout

alex-nov-git opened the pull request:
Added:
1. Tests for case:
- New Email; Billing same as Shipping (bad zipcode used) No Verification Suggestion chosen
- New Email; Billing same as Shipping (bad zipcode used) Choose a Verification Suggestion
- New Email; Billing (bad zipcode used) No Verification Suggestion chosen; Shipping (good)
- New Email; Billing (bad zipcode used) Choose a Verification Suggestion; Shipping (good)
- New Email; Billing (good) and Shipping (bad zipcode used) No Verification Suggestion chosen
- New Email; Billing (good) and Shipping (bad zipcode used) Choose a Verification Suggestion
- New Email; Billing same as Shipping (good address used); CC 4000000000000002 (expected card declined error)
2. Data provider, which provides settings and methods for test
3. Function, which provides testing created order address.
4. Recovery extension settings after test
5. Tests for one-step:
- New Email; Billing same as Shipping (good address used); CC 4000000000000002 (expected card declined error)
- New Email; Billing (good address used); Shipping (good address used); CC 4242424242424242
- Existing Email; Billing (good address used); Shipping (good address used); CC 4242424242424242

alex-nov-git committed:
Add SauceLabs functional tests for multi-page checkout

Added:
1. Tests for case (without final assert for created order address):
- New Email; Billing same as Shipping (bad zipcode used) No Verification Suggestion chosen
- New Email; Billing same as Shipping (bad zipcode used) Choose a Verification Suggestion
- New Email; Billing (bad zipcode used) No Verification Suggestion chosen; Shipping (good)
- New Email; Billing (bad zipcode used) Choose a Verification Suggestion; Shipping (good)
- New Email; Billing (good) and Shipping (bad zipcode used) No Verification Suggestion chosen
- New Email; Billing (good) and Shipping (bad zipcode used) Choose a Verification Suggestion
- New Email; Billing same as Shipping (good address used); CC 4000000000000002 (expected card declined error)
2. Data provider, which provides settings and methods for test

re issues: merchantprotocol/PHP-Sausage-Selenium#5, merchantprotocol/relieffactor.com#328

alex-nov-git committed:
Add SauceLabs functional tests for one-step checkout

Added:
Function, which provides testing created order address.
Recovery extension settings after test
Tests for one-step:
- New Email; Billing same as Shipping (good address used); CC 4000000000000002 (expected card declined error)
- New Email; Billing (good address used); Shipping (good address used); CC 4242424242424242
- Existing Email; Billing (good address used); Shipping (good address used); CC 4242424242424242

re issues: merchantprotocol/PHP-Sausage-Selenium#5, merchantprotocol/relieffactor.com#328

jonathonbyrdziak commented on tests/M1_Checkout.php:
```diff
@@ -0,0 +1,837 @@
+        $this->select($this->byId('cryozonic_stripe_expiration'))
+            ->selectOptionByValue(1);
+        $this->select($this->byId('cryozonic_stripe_expiration_yr'))
+            ->selectOptionByValue(2020);
```
`date('Y', strtotime('+1 year'))` should replace 2020

jonathonbyrdziak commented on tests/M1_Checkout.php:
```diff
@@ -0,0 +1,837 @@
+
+        // Use keys for select
+        $this->keys(PHPUnit_Extensions_Selenium2TestCase_Keys::DOWN);
+        $this->keys(PHPUnit_Extensions_Selenium2TestCase_Keys::ENTER);
```
The billing and the shipping forms both have Google Place Suggestions. Seems that we're testing the address suggestions on billing, but not on shipping. We should consider testing the google address suggestions on both forms. We should also consider what we might gain by testing non-auto suggest addresses in both forms.

jonathonbyrd committed:
Update config.staging.php

jonathonbyrd committed:
Update Makefile

jonathonbyrdziak commented:
Resolve the conflicts and then you're good to merge

alex-nov-git committed:
Merge branch 'master' into 5_M1_Checkout

alex-nov-git committed:
Replace hard coded date and decrease number of running tests

re issues: merchantprotocol/PHP-Sausage-Selenium#5, merchantprotocol/relieffactor.com#328
